### ISSUE TITLE
[Discussion] Training to recognise a specific song?

### ISSUE THREAD

**Issue by GelflingHand** [deezer/spleeter#538]
Hello! I'm interested in using Spleeter to separate a known song from the desired audio. Eg, if the song was playing on the radio in the next room when I made a recording of my own. What sort of dataset and configuration settings would get the best results?

Creating a dataset of separated files is easy enough - get a clean copy of the song, and mix it with a variety of different other recordings. But how many different recordings does it need to train well? And should I use the exact same song file each time, or would it help to use a variety of modified versions? Eg
- Doubled mono instead of real stereo (ie remove differences between left and right channels)?
- Slightly raised or lowered pitch?
- Slightly sped up or slowed down?
- Different volumes?
- Different sample rates?
- Breaking the song into shorter pieces rather than using the whole thing every time?
- Adding a bit of reverb or echo?

And are there any options in the configuration file that could help optimize Spleeter for recognising a specific song rather than an instrument? Eg, frame_length, frame_step, T, F, n_chunks_per_song, separation_exponent, mask_extension, learning_rate, batch_size - is changing any of the default options likely to help?

**Comment by romi1502** [comment]
Hi @GelflingHand 
From what I understand, you want to remove known music from a mix that contains this music (possibly a bit transformed) and dialogs/FX, such as a movie soundtrack.
If you're only interested in recovering speech and the music contains no vocals, you can try straighforwardly the trained models of spleeter and check the extracted vocal part.
Otherwise, you'd probably better use a model that takes advantage of the known audio signal in addition to the recording. This is something that spleeter was not designed for (only the mix is provided as input to the model in spleeter).

**Comment by GelflingHand** [comment]
Yes, you understand correctly. I'm interested in the general principle, so music may or may not contain vocals overlapping dialog. I've had promising results training my own model (better than the default model), but training takes a long time and I don't really know what I'm doing, so some guidance would be helpful.

The [wiki](https://github.com/deezer/spleeter/wiki/2.-Getting-started#train-model) includes a sample [config file](https://github.com/deezer/spleeter/blob/master/configs/musdb_config.json), but there is no explanation of the options in the file or what effects changing them might have. A few are self-explanatory, but I'm sure I'd not the only user who'd appreciate further documentation of the others.

It would also be helpful to know more about what how Spleeter interprets the dataset of separated files when training. Is it necessary to include variations of the song for Spleeter to recognise slight differences, or will it catch anything that's "close enough"? Does volume within the mix matter? Is the "other" stem just anything that doesn't seem to fit the first stem, or is it trying to decide which stem each frequency is a better match for (in which case the closer the "other" training set is to the dialog I'm trying to isolate, the better)?

Would it be better to break the song up in to shorter segments and train multiple models, eg verse, chorus, and bridge, or is Spleeter capable of making use of changes in the song over its full duration to help recognise it?

Thanks!
